data/src/test/java/com/example/data/settings/BaseSettingsRepositoryTest.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.data.settings
+
+import com.example.data.dashboard.cache.CurrencyPairEntity
+import com.example.data.dashboard.cache.FavoriteCurrenciesCacheDataSource
+import com.example.data.loadcurrencies.cache.CurrenciesCacheDataSource
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class BaseSettingsRepositoryTest {
+
+    private lateinit var repository: BaseSettingsRepository
+    private lateinit var currenciesCacheDataSource: FakeReadCurrenciesCacheDataSource
+    private lateinit var favoriteCurrenciesCacheDataSource: FakeFavoriteCurrenciesCacheDataSource
+
+    @Before
+    fun setUp() {
+        currenciesCacheDataSource = FakeReadCurrenciesCacheDataSource()
+        favoriteCurrenciesCacheDataSource = FakeFavoriteCurrenciesCacheDataSource()
+        repository = BaseSettingsRepository(
+            currenciesCacheDataSource = currenciesCacheDataSource,
+            favoriteCurrenciesCacheDataSource = favoriteCurrenciesCacheDataSource
+        )
+    }
+
+    @Test
+    fun testAllCurrencies(): Unit = runBlocking {
+        val currencies = repository.allCurrencies()
+        assertEquals(listOf("1", "2", "3", "4", "5"), currencies)
+    }
+
+    @Test
+    fun testSave(): Unit = runBlocking {
+        repository.save(from = "JPY", to = "EUR")
+
+        favoriteCurrenciesCacheDataSource.checkSavedCurrencyPairs(
+            pairs = listOf(
+                CurrencyPairEntity(
+                    fromCurrency = "JPY",
+                    toCurrency = "EUR",
+                    rates = -1.0,
+                    date = ""
+                )
+            )
+        )
+
+        repository.save(from = "JPY", to = "USD")
+
+        favoriteCurrenciesCacheDataSource.checkSavedCurrencyPairs(
+            pairs = listOf(
+                CurrencyPairEntity(
+                    fromCurrency = "JPY",
+                    toCurrency = "EUR",
+                    rates = -1.0,
+                    date = ""
+                ),
+                CurrencyPairEntity(
+                    fromCurrency = "JPY",
+                    toCurrency = "USD",
+                    rates = -1.0,
+                    date = ""
+                )
+            )
+        )
+    }
+
+    @Test
+    fun testAvailableDestinationsAndSave(): Unit = runBlocking {
+        var availableCurrencies = repository.availableDestinations("1")
```

Comment:
1. available destinations for USD : all currencies except USD
2. save USD to JPY
3. available destinations for USD : all currencies except USD and JPY
4. save EUR to JPY
5. avilable destinations for EUR : all currencies except EUR and JPY